rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -996,22 +996,22 @@ rmw_create_publisher(
       }
     }
     if (ddts) {
-      // Cannot unregister type here, in case another topic is using the same type.
+      // TODO Cannot unregister and free type here, in case another topic is using the same type.
```

Comment:
Yes, we currently have no way to decide when to destroy a `Topic`. Therefore we can't unregister the `DDSDynamicDataTypeSupport` which is bound to the topic and therefore we can't destroy the `DDSDynamicDataTypeSupport`. That memory is not even recovered when the node is being destroyed but lost permanently.